src/os/bluestore/BitmapFreelistManager.cc

```diff
@@ -221,6 +221,9 @@ bool BitmapFreelistManager::enumerate_next(uint64_t *offset, uint64_t *length)
 {
   std::lock_guard<std::mutex> l(lock);
 
+  // Don't store this in the object, iterators should be short lived.
+  KeyValueDB::Iterator enumerate_p;
```

Comment:
Ah, you are right. I wasn't looking closely enough at this, just the case where the offset and bl_pos are 0.  I'll rework to capture the existing behavior.